packager/media/codecs/h264_parser.cc

```diff
@@ -112,6 +112,17 @@ bool H264SliceHeader::IsSISlice() const {
     }                                                                      \
   } while (0)
 
+#define READ_LONG_OR_RETURN(out)                                           \
+  do {                                                                     \
+    long _out;                                                             \
+    int _tmp_out;                                                          \
+    READ_BITS_OR_RETURN(16, &_tmp_out);                                    \
```

Comment:
The template for ReadBits can handle up to 64 bits.  I think you could simplify this into one 32-bit call.